pkg/db/connection.go

```diff
@@ -32,103 +28,79 @@ type Settings struct {
 	PrefixedTables     bool          `cfg:"db_table_prefixed"`
 }
 
-type Connection struct {
-	kernel.BackgroundModule
+var defaultConnection = struct {
+	lck      sync.Mutex
+	instance *sqlx.DB
+	err      error
+}{}
 
-	logger   mon.Logger
-	settings Settings
-	db       *sqlx.DB
-
-	stop chan struct{}
-}
+func ProvideDefaultConnection(config cfg.Config, logger mon.Logger) (*sqlx.DB, error) {
+	defaultConnection.lck.Lock()
+	defer defaultConnection.lck.Unlock()
 
-var barrier = sync.WaitGroup{}
-var DefaultConnection *sqlx.DB = nil
-
-func NewConnection() *Connection {
-	barrier.Add(1)
-
-	connection := Connection{
-		stop: make(chan struct{}),
+	if defaultConnection.err != nil {
```

Comment:
Caching the err upon first connection prevents any kind of retry logic. I suggest not to do so.